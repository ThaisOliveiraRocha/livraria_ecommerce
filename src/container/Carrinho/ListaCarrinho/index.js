import ListaCarrinho from './ListaCarrinho';

export default ListaCarrinho;
