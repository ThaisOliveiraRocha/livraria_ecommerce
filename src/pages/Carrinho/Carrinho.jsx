import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
//import { Test } from './Carrinho.styles';

const Carrinho = (props) => (
  <div className="CarrinhoWrapper">
    Test content
  </div>
);

Carrinho.propTypes = {
  // bla: PropTypes.string,
};

Carrinho.defaultProps = {
  // bla: 'test',
};

const mapStateToProps = state => ({
  // blabla: state.blabla,
});

const mapDispatchToProps = dispatch => ({
  // fnBlaBla: () => dispatch(action.name()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Carrinho);
