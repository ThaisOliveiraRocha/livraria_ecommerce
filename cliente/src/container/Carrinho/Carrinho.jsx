import React, { useState } from "react";
import Layout from "../../components/Layout";
import {
  TableCart,
  TituloTopo,
  Cabecalho,
  BodyCart,
  ButtonContainer,
  TotalContainer
} from "./Carrinho.styles";
import { connect } from "react-redux";
import Button from "../../components/Button";
import Titulo from "../../components/Title";
import Text from "../../components/Text";
import ListaCarrinho from "./components/ListaCarrinho";
import ModalComponent from "../../components/ModalComponent";
import { removeLivro, showModal, showCompra } from "../../store/actions";
import { withRouter } from "react-router-dom";

const somaCarrinho = carrinho => {
  let soma = 0;
  carrinho.map(item => {
    soma += item.qtd * item.preco;
  });
  return soma.toFixed(2);
};

export const Carrinho = ({
  carrinho,
  history,
  funcaoConfirma,
  funcaoFinalizar,
  cancel,
  showRemoveModal,
  showModalCompra,
  showCompra
}) => {

  if (carrinho.length === 0) {
    {
      history.push("/");
    }
  }
  return (
    <Layout>
      {showRemoveModal && (
        <ModalComponent
          temaModal={"Deseja realizar a exclusão?"}
          mensagem={"O item selecionado será excluído permanentemente."}
          onClose={cancel}
          funcaoConfirma={funcaoConfirma}
          cancel={cancel}
        />
      )}
      {showModalCompra && (
        <ModalComponent
          temaModal={"Deseja finalizar a compra?"}
          mensagem={"Você será redirecionado para a Home."}
          onClose={() => showCompra(false)}
          funcaoConfirma={funcaoFinalizar}
          cancel={() => showCompra(false)}
        />
      )}
      <BodyCart>
        <TableCart>
          <thead>
            <TituloTopo>
              <Cabecalho>
                <Titulo>Título</Titulo>
              </Cabecalho>
              <Cabecalho>
                <Titulo>Quantidade</Titulo>
              </Cabecalho>
              <Cabecalho>
                <Titulo>Preço</Titulo>
              </Cabecalho>
              <Cabecalho>
                <Titulo>Descartar</Titulo>
              </Cabecalho>
            </TituloTopo>
          </thead>
          <tbody>
            {carrinho.map((item, index) => {
              return <ListaCarrinho key={index} item={item} />;
            })}
          </tbody>
        </TableCart>
        <TotalContainer>
          <Titulo>Total da compra:</Titulo>
          <Text> R$ {somaCarrinho(carrinho)}</Text>
        </TotalContainer>
        <ButtonContainer>
          <Button onClick={() => showCompra(true)} className="btnFinalizar">
            <Text color="white">Finalizar Compra</Text>
          </Button>
          <Button background="#818A96" onClick={() => history.push("/home")} className="btnCancelar">
            <Text color="white">Cancelar</Text>
          </Button>
        </ButtonContainer>
      </BodyCart>
    </Layout>
  );
};

Carrinho.propTypes = {
  // bla: PropTypes.string,
};

const mapStateToProps = state => ({
  carrinho: state.livraria.carrinho,
  showRemoveModal: state.livraria.removeModal,
  showModalCompra: state.livraria.modalCompra
});

const mapDispatchToProps = (dispatch, { history }) => ({
  funcaoConfirma: () => {
    dispatch(removeLivro());
  },
  funcaoFinalizar: () => {
    history.push("/");
  },
  cancel: () => {
    dispatch(showModal("", false));
  },
  showCompra: (status) => {
    dispatch(showCompra(status));
  }
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Carrinho));
