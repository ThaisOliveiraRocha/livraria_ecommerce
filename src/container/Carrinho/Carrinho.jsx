import React, { PureComponent } from "react";
import ListaCarrinho from "./components/ListaCarrinho/ListaCarrinho";
import Layout from '../../components/Layout';
// import {} from "./Carrinho.styles";

class Carrinho extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      hasError: false
    };
  }

  render() {
    if (this.state.hasError) {
      return <h1>Something went wrong.</h1>;
    }
    return (
      <Layout>
        <ListaCarrinho />
      </Layout>
    );
  }
}

Carrinho.propTypes = {
  // bla: PropTypes.string,
};

Carrinho.defaultProps = {
  // bla: 'test',
};

export default Carrinho;
