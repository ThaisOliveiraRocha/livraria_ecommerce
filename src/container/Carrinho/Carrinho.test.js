import React from "react";
import { shallow } from "enzyme";
import Carrinho from "./Carrinho";

describe("<Carrinho />", () => {
  it("should render without crashing", () => {
    shallow(<Carrinho />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<Carrinho />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
