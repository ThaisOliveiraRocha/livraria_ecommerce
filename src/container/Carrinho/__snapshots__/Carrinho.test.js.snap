// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carrinho /> nao há itens no carrinho 1`] = `
<Layout>
  <styled.div>
    <styled.table>
      <thead>
        <styled.tr>
          <styled.th>
            <Title>
              Título
            </Title>
          </styled.th>
          <styled.th>
            <Title>
              Quantidade
            </Title>
          </styled.th>
          <styled.th>
            <Title>
              Preço
            </Title>
          </styled.th>
          <styled.th>
            <Title>
              Descartar
            </Title>
          </styled.th>
        </styled.tr>
      </thead>
      <tbody>
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "2018",
              "autor": "Colleen Hoover",
              "genero": "romance",
              "imagem": "book1.png",
              "preco": "40.00",
              "qtd": 0,
              "situacao": "novo",
              "titulo": "É Assim Que Acaba",
            }
          }
          key="0"
        />
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "2016",
              "autor": "Colleen Hoover",
              "genero": "drama",
              "imagem": "book2.png",
              "preco": "18.00",
              "qtd": 0,
              "situacao": "usado",
              "titulo": "Tarde Demais",
            }
          }
          key="1"
        />
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "2010",
              "autor": "William Gibson",
              "genero": "ficção cientifica",
              "imagem": "book3.png",
              "preco": "32.90",
              "qtd": 0,
              "situacao": "novo",
              "titulo": "Neuromancer",
            }
          }
          key="2"
        />
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "1990",
              "autor": "Stephen King",
              "genero": "terror",
              "imagem": "book4.png",
              "preco": "29.50",
              "qtd": 0,
              "situacao": "usado",
              "titulo": "It - A coisa",
            }
          }
          key="3"
        />
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "2000",
              "autor": "George Martin",
              "genero": "fantasia",
              "imagem": "book5.png",
              "preco": "25.00",
              "qtd": 0,
              "situacao": "novo",
              "titulo": "Game of Thrones",
            }
          }
          key="4"
        />
      </tbody>
    </styled.table>
    <styled.div>
      <Title>
        Total da compra:
      </Title>
      <Text>
         R$ 
        0.00
      </Text>
    </styled.div>
    <styled.div>
      <Button
        background="green"
        className="btnFinalizar"
        onClick={[Function]}
      >
        Finalizar
      </Button>
      <Button
        background="red"
        className="btnCancelar"
        onClick={[Function]}
      >
        Cancelar
      </Button>
    </styled.div>
  </styled.div>
</Layout>
`;

exports[`<Carrinho /> should render basic infos 1`] = `
<Layout>
  <styled.div>
    <styled.table>
      <thead>
        <styled.tr>
          <styled.th>
            <Title>
              Título
            </Title>
          </styled.th>
          <styled.th>
            <Title>
              Quantidade
            </Title>
          </styled.th>
          <styled.th>
            <Title>
              Preço
            </Title>
          </styled.th>
          <styled.th>
            <Title>
              Descartar
            </Title>
          </styled.th>
        </styled.tr>
      </thead>
      <tbody>
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "2018",
              "autor": "Colleen Hoover",
              "genero": "romance",
              "imagem": "book1.png",
              "preco": "40.00",
              "qtd": 0,
              "situacao": "novo",
              "titulo": "É Assim Que Acaba",
            }
          }
          key="0"
        />
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "2016",
              "autor": "Colleen Hoover",
              "genero": "drama",
              "imagem": "book2.png",
              "preco": "18.00",
              "qtd": 0,
              "situacao": "usado",
              "titulo": "Tarde Demais",
            }
          }
          key="1"
        />
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "2010",
              "autor": "William Gibson",
              "genero": "ficção cientifica",
              "imagem": "book3.png",
              "preco": "32.90",
              "qtd": 0,
              "situacao": "novo",
              "titulo": "Neuromancer",
            }
          }
          key="2"
        />
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "1990",
              "autor": "Stephen King",
              "genero": "terror",
              "imagem": "book4.png",
              "preco": "29.50",
              "qtd": 0,
              "situacao": "usado",
              "titulo": "It - A coisa",
            }
          }
          key="3"
        />
        <withRouter(Connect(ListaCarrinho))
          item={
            Object {
              "ano": "2000",
              "autor": "George Martin",
              "genero": "fantasia",
              "imagem": "book5.png",
              "preco": "25.00",
              "qtd": 0,
              "situacao": "novo",
              "titulo": "Game of Thrones",
            }
          }
          key="4"
        />
      </tbody>
    </styled.table>
    <styled.div>
      <Title>
        Total da compra:
      </Title>
      <Text>
         R$ 
        0.00
      </Text>
    </styled.div>
    <styled.div>
      <Button
        background="green"
        className="btnFinalizar"
        onClick={[Function]}
      >
        Finalizar
      </Button>
      <Button
        background="red"
        className="btnCancelar"
        onClick={[Function]}
      >
        Cancelar
      </Button>
    </styled.div>
  </styled.div>
</Layout>
`;
